### PR TITLE
Migrate to CP preinstall hooks

### PR DESCRIPTION
**Pull request by orta**
This reds locally, but works when I'm doing it with an app. Wanted this available for people to use as it's the fix for: #124

### CODE REVIEW

**Comment by ashfurrow** [comment]
Cool, looks great! Merge whenevs.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## 1.6.0
```
Comment (marcelofabri) [inline CHANGELOG.md]: Version bump needed here

File: lib/plugin.rb [hunk 2]
```diff
@@ -6,6 +6,11 @@ module CocoaPodsKeys
   class << self
     include FileUtils
 
+    # Register for the pre-install hooks to setup & run Keys
+    Pod::HooksManager.register('cocoapods-keys', :pre_install) do
+      CocoaPodsKeys.setup
```
Comment (segiddins) [inline lib/plugin.rb]: this should now call `setup` with the options, instead of pulling off of `Pod::Config.instance.podfile`

**Comment by orta** [comment]
OK, addressed all feedback, tests are green ( and rubocop is happy ) - I've run this as a `:path` gem successfully against Eigen of this PR https://github.com/artsy/eigen/pull/1383

**Pull request merged**